webapp/src/components/shareBoard/shareBoard.tsx

```diff
@@ -294,11 +300,12 @@ export default function ShareBoardDialog(props: Props): JSX.Element {
                     <div className='share-input'>
                         <SearchIcon/>
                         <Select
+                            key={JSON.stringify(membersCount)}
```

Comment:
This forces the Async Select list to reload. I tried wrapping `searchAndFilterTeamUsers` with useCallback() but could not get it to work. I'm open to other suggestions. This is the solution suggested [here](https://github.com/JedWatson/react-select/issues/1581).

Without this, when a user is added. If you click on the drop down list again the user is still listed. Once you perform a search, the list is updated.

Reply:
Why not use the `filterOptions` property of the `Select` component? looks like the right approach here.

Reply:
Way better...Thanks.